[[ 22635 ]] Remove unsupported Android minimum deployment versions

livecodepanos opened the pull request:
Closes https://quality.livecode.com/show_bug.cgi?id=22635

The android engines are compiled with `--api 21` thus the minimum supported Android version is now API 21 (Android 5.0).

Note that if an existing stack has the minimum Android version set to something that was removed by this patch, this is not a problem.

The S/B will present an dialog informing the user about the new minimum Android version supported, and ask them if they want to update their existing minimum Android version (see `updateMinimumAPI`)

montegoulding reviewed (changes requested):
@livecodepanos actual changes look good. There's just a few unnecessary whitespace changes that should be dropped.

livecodepanos committed:
[[ 22635 ]] Remove unsupported Android minimum deployment versions

The android engines are compiled with `--api 21` thus the minimum
supported Android version is now API 21 (Android 5.0).

livecodepanos commented:
@montegoulding done :)

montegoulding commented:
@livecode-vulcan review ok 4d4c759

livecode-vulcan commented:
💙 review by @montegoulding ok 4d4c7591bfcd7dd002369050721eeb0102a6f069

livecode-vulcan commented:
😎 test success 4d4c7591bfcd7dd002369050721eeb0102a6f069

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/749)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/158)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/158)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/158)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/158)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2275)
* try-community-universal-ios-iphoneos13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/9)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/9)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2304)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2291)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2290)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2325)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2265)
